### PR TITLE
Created a new type of question URL to make submissions in that catego…

### LINKED ISSUE #119: URL turned automatically to links
Instructors often ask questions whose answers are URLs.
Could those be turned automatically to links?

### PR DESCRIPTION
**Pull request by brendon-lind**
…ry link in a new tab

# A brief description of the purpose of the changes contained in this PR.
Added new URL Custom Question to define homework.
Added this to submit homework page.
Added this to Grade Page to make links clickable to a new tab.

# Known issues to be addressed in a separate PR
Resolves #119 

# A checklist for hand testing
- [ ] Create a new custom question using URL as its Type.
- [ ] Submit or resubmit homework on that assignment including a URL in that newly created question.
- [ ] Go Grade that submission and click the URL link.

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [x] Code review by reviewer
- [x] Hand tested by reviewer
- [ ] Ready to merge

### CODE REVIEW

**Comment by gibsonbailey** [comment]
I created a migration for you.

File: static/riotjs/wizard/grade_homework.tag [hunk 1]
```diff
@@ -4,7 +4,10 @@
         <ol>
             <li each="{ question in definition.custom_questions }">{ question.question }
                 <ul>
-                    <li each="{ answer in question.student_answers }">{ answer }</li>
+                    <li if="{ question.question_type == 'UL' }" each="{ answer in question.student_answers }">
+                        <a href="{ answer }" target="_blank">{ answer }</a>
+                    </li>
+                    <li if="{ question.question_type != 'UL' }" each="{ answer in question.student_answers }">{ answer }</li>
                 </ul>
```
Comment (gibsonbailey) [inline static/riotjs/wizard/grade_homework.tag:11]: I am not seeing the student's answers to these questions in the grading view here. Let me know if you see the same behavior. I'm experiencing the same on the submission detail page. I can only see questions, and not the responses.

File: static/riotjs/student/submit_homework.tag [hunk 2]
```diff
@@ -175,6 +175,20 @@
...
+
+                <div class="ui labeled input">
+                    <div class="ui label">
+                        URL
+                    </div>
+                    <input data-question-id="" name="{'question_answer_' + index}" ref="{'question_answer_' + question.id}"
+                          type="text" value="{question.prev_answer || ''}" placeholder="mysite.com">
+                </div>
+            </div>
+
```
Comment (gibsonbailey) [inline static/riotjs/student/submit_homework.tag:191]: We ought to implement validation of the response as a valid url format here. You should write the validation in the serializer for the definition.

File: static/riotjs/wizard/grade_homework.tag (hunk 1 above)
Comment (gibsonbailey) [reply static/riotjs/wizard/grade_homework.tag:11]: Also, the `homework_answers.html` page should display these as links. Right now, I don't believe I can see the answer.

**Review by gibsonbailey — CHANGES_REQUESTED** [review]
Good work. Not seeing the answers when grading or when looking at the submission detail page. Once those issues are taken care of, it should be good to go!

File: static/riotjs/student/submit_homework.tag (hunk 2 above)
Comment (brendon-lind) [reply static/riotjs/student/submit_homework.tag:191]: Thanks. Validation functioning now.

File: static/riotjs/wizard/grade_homework.tag (hunk 1 above)
Comment (brendon-lind) [reply static/riotjs/wizard/grade_homework.tag:11]: Bug fixed good catch.

**Pull request merged**